capellambse/extensions/validation/rules.py

```diff
@@ -0,0 +1,430 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capellambse contributors
+# SPDX-License-Identifier: Apache-2.0
+from __future__ import annotations
+
+import re
+import typing as t
+
+from capellambse.model import common as c
+from capellambse.model.crosslayer import fa
+from capellambse.model.layers import ctx as sa
+
+from . import _validate
+
+
+@_validate.register_rule(
+    category=_validate.Category.REQUIRED,
+    types=[sa.SystemComponent, sa.SystemFunction, sa.Capability],
+    id="Rule-001",
+    name="Object has a description or summary",
+    rationale=(
+        "A comprehensive description or summary for an object is "
+        "essential to ensure a clear understanding of its purpose, "
+        "function, and role within the system. Providing a concise, "
+        "yet informative description fosters better collaboration "
+        "among team members, reduces ambiguity, and facilitates "
+        "efficient decision-making."
+    ),
+    action="fill the description and/or summary text fields",
+    applicable_to="SystemComponent where <i>is_actor</i> is set to True.",
+)
+def has_non_empty_description_or_summary(
+    obj: c.GenericElement,
+) -> bool | t.Literal["NotApplicable"]:
+    # if not obj.is_actor:  # Precondition
```

Comment:
due to separation of concern and SRP the rule function should do just one thing namely check the compliance of an object with the rule. The selection of the objects the rule is applicable to is a separate job. "Not applicable" is actually an anti-pattern. as a class with just static methods would be, too. So an additional `fllter` decorator parameter, which lets through just the objects the rule is applicable to is still more appropriate, imho.